chain-impl-mockchain/src/ledger/evm.rs

```diff
@@ -31,19 +92,68 @@ impl EvmState for super::Ledger {
         &self.evm.environment
     }
 
-    fn account(&self, address: Address) -> Option<Account> {
-        self.evm.accounts.get(&address).cloned()
+    fn account(&self, evm_address: &EvmAddress) -> Option<EvmAccount> {
+        match self.evm.address_mapping.jor_address(evm_address) {
+            Some(jor_address) => match self.accounts.get_state(jor_address) {
+                Ok(account) => Some(EvmAccount {
+                    balance: account.value.0.into(),
+                    state: account.evm_state.clone(),
+                }),
+                Err(_) => None,
+            },
+            None => None,
+        }
     }
 
-    fn contains(&self, address: Address) -> bool {
-        self.evm.accounts.contains(&address)
+    fn contains(&self, address: &EvmAddress) -> bool {
+        self.evm.address_mapping.jor_address(address).is_some()
     }
 
-    fn modify_account<F>(&mut self, address: Address, f: F)
+    fn modify_account<F>(&mut self, address: EvmAddress, f: F) -> Result<(), ExitError>
     where
-        F: FnOnce(Account) -> Option<Account>,
+        F: FnOnce(EvmAccount) -> Option<EvmAccount>,
     {
-        self.evm.accounts = self.evm.accounts.clone().modify_account(address, f);
+        let address = self
+            .evm
+            .address_mapping
+            .jor_address(&address)
+            .ok_or_else(|| {
+                ExitError::Other(
```

Comment:
we can provide a better typed error here

Reply:
Actually no, `ExitError` it is a `evm-library` error type, so it is the only way to provide a custom error message.
Or do you mean something else ?

Reply:
Ah sorry, forget this then